### PR TITLE
Final Version

### PR DESCRIPTION
**Pull request by SidFerreira**
Simple app:

- Route List screen:  allow to search for a street name and lists the routes for this street.
- Route Details: 
 - Tab Stops: lists each stop of this route 
 - Tab Departures: Lists for weekdays, sundays and saturdays the time for each departure of the route.

### CODE REVIEW

File: www/js/controllers.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+.controller('RouteDetailCtrl', function($scope, $stateParams, routeService) {
+  var routeId = $stateParams.routeId;
+  $scope.stops = undefined;
+
+  //It is cached, it was queried on the previous screen
+  var route = routeService.getRouteInfo(routeId);
+  if(route) {
+    $scope.route = route;
+    /****************/
+    routeService.findStopsByRouteId(routeId).success(function(data, status, headers, config) {
```
Comment (arctouch-andrecardoso) [inline www/js/controllers.js]: Why `routeService#findStopsByRouteId` returns data, status, headers and config?

File: www/js/controllers.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      $scope.stops = data.rows;
+    }).
+    error(function(data, status, headers, config) {
+      showAlert("Couldn't find the stops for this route.");
+    });
+
+    /****************/
+    routeService.findDeparturesByRouteId(routeId).success(function(data, status, headers, config) {
+      var departures = {};
```
Comment (arctouch-andrecardoso) [inline www/js/controllers.js]: Could this "departure grouping" code be located in the service class?

File: www/js/services.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            'Content-Type': 'application/json',
+            'Authorization': 'Basic V0tENE43WU1BMXVpTThWOkR0ZFR0ek1MUWxBMGhrMkMxWWk1cEx5VklsQVE2OA==',
+            'X-AppGlu-Environment': "staging"
+          },
+          data: { "params": params }
+        };
+
+    $http(request).success(function(data) {
+      deferred.resolve(data);
+      if(apiName == 'findRoutesByStopName') {
```
Comment (arctouch-andrecardoso) [inline www/js/services.js]: Why you do this check here? Couldn't this part of the code be located directly in the `findRoutesByStopName` method?

File: www/js/services.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        if(data.rows.length > 0) {
+          for (var i = data.rows.length - 1; i >= 0; i--) {
+            var route = data.rows[i];
+            cacheRoutes[route.id] = route;
```
Comment (arctouch-andrecardoso) [inline www/js/services.js]: Why caching the results?

**Comment by arctouch-andrecardoso** [comment]
Reviewed. Please take a look at my comments. Thanks!

**Comment by SidFerreira** [comment]
Hi Andre!
Follows some comments:

- "Why `routeService#findStopsByRouteId` returns data, status, headers and config?"
 - I indeed forgot the parameters there. I started a refactory with one thing in mind and changed it later, but didn't cleaned up the controllers.
- "Could this "departure grouping" code be located in the service class?"
 - Indeed it could. But at same time, this grouping is a logic from this controller, not from the method.
- "Why you do this check here? Couldn't this part of the code be located directly in the `findRoutesByStopName` method?" (Also "Why caching the results?")
 - It wasn't clear to me which check you meant. I believe it is the line 20, if so, this test is used for caching the route IDs. This is used to retrieve the route information on the details screen (instead of sending a whole object). 

Im fixing this and improving a bit the code. If you have any extra comment, please go ahead.

**Comment by SidFerreira** [comment]
https://github.com/SidFerreira/busRouteClientJS/pull/2

**Pull request closed without merging**